.github/actions/install-deps/action.yml

```diff
@@ -0,0 +1,98 @@
+    required: true
+    default: '4.210'
+  verilator-path:
+    description: Path at which to install Veriltator
+    required: true
+    default: /tmp/verilator
+  verible-version:
+    description: Verible version to install
+    required: true
+    default: 'v0.0-3622-g07b310a3'
+  verible-path:
+    description: Path at which to install Verible
+    required: true
+    default: /tmp/verible
+  apt-archive-path:
+    description: Path at which to download APT package archives
+    required: true
+    default: /tmp/apt-archives
+
+runs:
+  using: composite
+  steps:
+    - name: Enable caching for APT packages
+      uses: actions/cache@v4
+      with:
+        # We cache the directory where APT downloads `.deb` archives for packages.
+        path: ${{ inputs.apt-archive-path }}
+        key: ${{ runner.os }}-${{ hashFiles('apt-requirements.txt') }}
+      id: cache-apt
+
+    - name: Install system dependencies
+      run: |
+        mkdir -p "${{ inputs.apt-archive-path }}"
+        # Download archives separately, otherwise APT deletes them once installed.
+        grep '^[^#]' apt-requirements.txt | \
+          xargs sudo apt -o Dir::Cache::archives=${{ inputs.apt-archive-path }} install --download-only
+        grep '^[^#]' apt-requirements.txt | \
+          xargs sudo apt -o Dir::Cache::archives=${{ inputs.apt-archive-path }} install -y
+        # Allow the cache-saving step to read the directory contents.
+        sudo chmod -R +rw ${{ inputs.apt-archive-path }}
+      shell: bash
+
+    - uses: actions/setup-python@v5
+      with: # TODO: use `python-version-file` when we have `pyproject.toml`.
+        python-version: 3.9
+        cache-dependency-path: python-requirements.txt
+        cache: pip
+
+    - name: Install Python dependencies
+      run: python3 -m pip install -r python-requirements.txt --require-hashes
+      shell: bash
+
+    - name: Enable caching for EDA tools
+      uses: actions/cache@v4
+      with:
+        path: |
+          ${{ inputs.verilator-path }}
+          ${{ inputs.verible-path }}
+        key: ${{ runner.os }}-${{ inputs.verilator-version }}-${{ inputs.verible-version }}
+      id: cache-eda
+
+    - name: Install Verilator
+      if: ${{ steps.cache-eda.outputs.cache-hit != 'true' }}
+      run: |
+        VERILATOR_TAR="verilator-v${{ inputs.verilator-version }}.tar.gz"
+        VERILATOR_URL="https://storage.googleapis.com/verilator-builds/${VERILATOR_TAR}"
+        curl -f -Ls -o "/tmp/${VERILATOR_TAR}" "$VERILATOR_URL" || {
+          echo "failed to download Verilator from ${VERILATOR_URL}" >&2
+          exit 1
+        }
+        mkdir -m 777 -p "${{ inputs.verilator-path }}"
+        tar -C "${{ inputs.verilator-path }}" -xvzf "/tmp/${VERILATOR_TAR}" --strip-components=1
+        echo "${{ inputs.verilator-path }}/bin" >> "$GITHUB_PATH"
+      shell: bash
+
+    - name: Install Verible
+      if: ${{ steps.cache-eda.outputs.cache-hit != 'true' }}
```

Comment:
This switches from downloading from GitHub release to GitHub cache. Probably not worth doing it?